cli/daemon/run/runtime_config.go

```diff
@@ -0,0 +1,411 @@
+package run
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/netip"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/rs/xid"
+
+	encore "encore.dev"
+	"encore.dev/appruntime/exported/config"
+	"encr.dev/pkg/option"
+	meta "encr.dev/proto/encore/parser/meta/v1"
+)
+
+const (
+	runtimeCfgEnvVar    = "ENCORE_RUNTIME_CONFIG"
+	appSecretsEnvVar    = "ENCORE_APP_SECRETS"
+	serviceCfgEnvPrefix = "ENCORE_CFG_"
+	listenEnvVar        = "ENCORE_LISTEN_ADDR"
+)
+
+type RuntimeEnvGenerator struct {
+	// DI interfaces
+
+	// The application to generate the config for
+	App interface {
+		PlatformID() string
+		PlatformOrLocalID() string
+	}
+
+	// The infra manager to use
+	InfraManager interface {
+		SQLConfig(db *meta.SQLDatabase) (config.SQLServer, config.SQLDatabase, error)
+		PubSubTopicConfig(topic *meta.PubSubTopic) (config.PubsubProvider, config.PubsubTopic, error)
+		PubSubSubscriptionConfig(topic *meta.PubSubTopic, sub *meta.PubSubTopic_Subscription) (config.PubsubSubscription, error)
+		RedisConfig(redis *meta.CacheCluster) (config.RedisServer, config.RedisDatabase, error)
+	}
```

Comment:
These are interfaces as this process is complicated enough I want to be able to mock them in tests. I just ran out of time to write the tests today.

It might also be useful to keep these interfaces outside of the needs of unit tests, so things like our CI/CD pipeline can reuse the EnvGenerator without needing to have modifications to the `ResourceManager` struct made